src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Selectors/SecurityTokenAuthenticator.cs

```diff
@@ -20,7 +21,7 @@ public bool CanValidateToken(SecurityToken token)
             return CanValidateTokenCore(token);
         }
 
-        public ReadOnlyCollection<IAuthorizationPolicy> ValidateToken(SecurityToken token)
+        public async Task<ReadOnlyCollection<IAuthorizationPolicy>> ValidateTokenAsync(SecurityToken token)
         {
```

Comment:
I've been thinking about needing to avoid sync over async and I think we'd actually benefit from using ValueTask instead. When the implementation is synchronous, it has zero extra allocation as ValueTask is a struct. When it goes async, it gets backed by a Task and has no additional allocation. Here's a [great blog post by Stephen Toub](https://devblogs.microsoft.com/dotnet/understanding-the-whys-whats-and-whens-of-valuetask/) about ValueTask, how to use it, and why to use it.

Reply:
update: I added a commit to remove sync over async from SpnegoTokenAuthenticator